engineapi/engineapi/contracts_actions.py

```diff
@@ -66,30 +114,39 @@ def validate_method_and_params(
             Web3.toChecksumAddress(parameters["signer"])
         except:
             raise InvalidAddressFormat("Parameter signer must be a valid address")
-        required_params["amount"] = str(required_params["amount"])
     else:
         raise ValueError(f"Unknown contract type {contract_type}")
 
 
 def register_contract(
     db_session: Session,
-    blockchain: str,
+    blockchain_name: str,
```

Comment:
Do we want to add lookup by chain ID as well? In that case, `blockchain_name` and `chain_id` would be `Optional` but not both can be `None`.

Reply:
I suggest to add it in next versions, also we can modify get contract endpoints with chain id filter/testnet filter, etc